src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceProgress.cs

```diff
@@ -166,6 +217,10 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
 
         private ProgressPane _progPane = null;
         private PendingProgress _pendingProgress = null;
+        // The timer set up 'progPaneUpdateFlag' every 'UpdateTimerThreshold' milliseconds to update 'ProgressPane'
+        private Timer _progPaneUpdateTimer;
```

Comment:
I think `_pendingProgress.Update` with a `completed` record will just clear up all structures for the corresponding progress activity. It won't really solve the potential `freeze` problem, but it's good to have.